02_knight_and_princess/dana_stepina/main.cpp

```diff
@@ -0,0 +1,174 @@
+        vertex_start(vertex_connection_start),
+        vertex_end(vertex_connection_end) {}
+};
+
+class Graph {
+ public:
+  void add_vertex();
+  void add_edge(const VertexId& from_vertex_id, const VertexId& to_vertex_id);
+  const std::vector<Edge>& get_edges() const { return edges_; }
+  const std::vector<Vertex>& get_vertices() const { return vertices_; }
+  bool has_vertex_id(const VertexId& vertex_id) const;
+  bool is_connected(const VertexId& from_vertex_id,
+                    const VertexId& to_vertex_id) const;
+
+ private:
+  std::vector<Vertex> vertices_;
+  std::vector<Edge> edges_;
+  EdgeId edge_id_counter_ = 0;
+  VertexId vertex_id_counter_ = 0;
+
+  VertexId get_new_vertex_id() { return vertex_id_counter_++; }
+  EdgeId get_new_edge_id() { return edge_id_counter_++; }
+};
+
+void Graph::add_vertex() {
+  vertices_.emplace_back(get_new_vertex_id());
+}
+void Graph::add_edge(const VertexId& from_vertex_id,
+                     const VertexId& to_vertex_id) {
+  assert(has_vertex_id(from_vertex_id) &&
+         "There is no such vertex in the graph.");
+  assert(has_vertex_id(to_vertex_id) &&
+         "There is no such vertex in the graph.");
+  assert(!is_connected(from_vertex_id, to_vertex_id) &&
+         "These vertices are already connected");
+
+  const auto& new_edge =
+      edges_.emplace_back(get_new_edge_id(), from_vertex_id, to_vertex_id);
+  vertices_[from_vertex_id].add_edge_id(new_edge.id);
+  vertices_[to_vertex_id].add_edge_id(new_edge.id);
+}
+bool Graph::has_vertex_id(const VertexId& vertex_id) const {
+  for (const auto& vertex : vertices_)
+    if (vertex_id == vertex.id)
+      return true;
+  return false;
+}
+bool Graph::is_connected(const VertexId& from_vertex_id,
+                         const VertexId& to_vertex_id) const {
+  assert(has_vertex_id(from_vertex_id) &&
+         "There is no such vertex in the graph.");
+  assert(has_vertex_id(to_vertex_id) &&
+         "There is no such vertex in the graph.");
+
+  for (const auto& edge_id_from_vertex :
+       vertices_[from_vertex_id].get_edge_ids())
+    for (const auto& edge_id_to_vertex : vertices_[to_vertex_id].get_edge_ids())
+      if (edge_id_from_vertex == edge_id_to_vertex)
+        return true;
+  return false;
+}
+
+//ВЫВОД В ФАЙЛ JSON
+std::string get_vertex_string(const Vertex& vertex) {
+  std::string str_vertex =
+      "\t\t{\"id\":" + std::to_string(vertex.id) + ",\"edge_ids\":[";
+  for (const auto& edge_id : vertex.get_edge_ids()) {
+    str_vertex += std::to_string(edge_id);
+    if (edge_id != vertex.get_edge_ids().back())
+      str_vertex += ",";
+  }
+  str_vertex += "]}";
+  return str_vertex;
+}
+std::string get_edge_string(const Edge& edge) {
+  std::string str_edge = "\t\t{\"id\":" + std::to_string(edge.id) +
+                         ",\"vertex_ids\":[" +
+                         std::to_string(edge.vertex_start) + "," +
+                         std::to_string(edge.vertex_end) + "]}";
+  return str_edge;

```

Comment:
Вроде писал уже тебе в прошлых ревью, что здесь нет смысла создавать переменную
```suggestion
  return "\t\t{\"id\":" + std::to_string(edge.id) +
          ",\"vertex_ids\":[" +
          std::to_string(edge.vertex_start) + "," +
          std::to_string(edge.vertex_end) + "]}";
```